### ISSUE TITLE
shell: "Administrative access" dialog is ugly with long prompts

### ISSUE THREAD

**Issue by mvollmer** [cockpit-project/cockpit#16696]
Cockpit version: 257
OS: all

The "Administrative access" dialog looks like this with long prompts:

![image](https://user-images.githubusercontent.com/3228183/145030465-d61d8b0a-1528-4796-acfd-9bb315cb3ae4.png)

(The above screenshot has been produced by hard coding the long prompt into `pkg/shell/superuser.jsx`.)

**Comment by mvollmer** [comment]
Also see #16688 for ideas of how to fix this.  #16688 is a more sever variant of this issue here on RHEL 8.5 with Cockpit 251, where a long prompt actually moves the text input field out of the dialog and thus makes the dialog inoperable.

**Comment by garrett** [comment]
> We could detect excessively long prompts (> 50 bytes) and display them as a message paragraph above the input field.

 Yes, that would be better.

I also suggested that if multiple hard enters are returned from sudo, then we may want to include some formatting by replacing them with `<br>`. (This is one of the very few times where `<br>` actually makes sense.) _Ideally_, we'd split on `/\n\n+/`, have each in parts in paragraphs, and change remaining `\n` to `<br>`. But simply changing `\n` to `<br>` would already be an improvement.)

 It gets a bit hard to read when it's all on one line, even after it wraps, I'm sure. Here's an example screenshot of text:

![image](https://user-images.githubusercontent.com/95657298/144929679-02472e18-6d0a-4022-8d99-32037758e273.png)

**Comment by mvollmer** [comment]
> I also suggested that if multiple hard enters are returned from sudo, then we may want to include some formatting by replacing them with <br>

We could also switch to using `pre` when we detect certain characters, like `\n` and `\t`. The presence of these characters probably means that the text was designed for a mono-space terminal, and we could simulate that.  That would make all kinds of text markup work, like

```
Enter one of these:

 1 - Blah, blah
     blah, bla

 2 - Bla, blah, blah
```

We display the "motd" in a `pre`, for example, to keep ASCII art in tact that is meant for a terminal.

**Comment by garrett** [comment]
> We could also switch to using pre when we detect certain characters, like \n and \t.

No, `<pre>` is not the right choice. We'd want the `white-space` CSS property, specifically `white-space: pre-wrap;` or `white-space: pre-line;` (this latter one strips out indentations, which could be good in some cases, but bad when there are preceding space before, including `\t`).

Why not `<pre>`? We don't want a monospaced font. This isn't HTML code. Newlines and indentations could be enough. This isn't the place for ASCII art.

**Comment by garrett** [comment]
We could also switch to a monospaced font if certain characters are found (ranges outside of the norm). But then we'd probably want these heuristics for the MOTD as well.

**Comment by garrett** [comment]
Can we solve this with a quick fix for the release and do a more proper fix for later?

I guess that would split the issue up into three parts:

1. **Backport** to make things work.
2. A **quick fix** (in this release, if possible) to make the text and form show up in a mostly reasonable manner.
3. A **more correct** fix that optimizes the display.
